packages/airnode-protocol-v1/contracts/monetization/RequesterAuthorizerWhitelisterWithTokenDeposit.sol

```diff
@@ -158,6 +189,73 @@ contract RequesterAuthorizerWhitelisterWithTokenDeposit is
                     false
                 );
         }
+    }
+
+    /// @notice Withdraws tokens previously deposited for the requester to be
+    /// whitelisted for the Airnode–endpoint pair on the chain
+    /// @param airnode Airnode address
+    /// @param chainId Chain ID
+    /// @param endpointId Endpoint ID
+    /// @param requester Requester address
+    function withdrawTokens(
+        address airnode,
+        uint256 chainId,
+        bytes32 endpointId,
+        address requester
+    ) external override onlyNonBlockedRequester(airnode, requester) {
+        TokenDeposits
+            storage tokenDeposits = airnodeToChainIdToEndpointIdToRequesterToTokenDeposits[
+                airnode
+            ][chainId][endpointId][requester];
+        uint256 tokenWithdrawAmount = tokenDeposits.depositorToAmount[
+            msg.sender
+        ];
+        require(tokenWithdrawAmount != 0, "Sender has not deposited tokens");
+        tokenDeposits.depositorToAmount[msg.sender] = 0;
+        uint256 earliestWithdrawalTime = tokenDeposits
+            .depositorToEarliestWithdrawalTime[msg.sender];
+        if (withdrawalLeadTime != 0) {
+            require(
+                earliestWithdrawalTime != 0,
+                "Withdrawal intent not signaled"
+            );
+        }
+        if (earliestWithdrawalTime != 0) {
+            require(
+                earliestWithdrawalTime <= block.timestamp,
+                "Not withdrawal time yet"
+            );
+            uint256 tokenDepositsCount = tokenDeposits.count;
+            emit WithdrewTokens(
+                airnode,
+                chainId,
+                endpointId,
+                requester,
+                msg.sender,
+                tokenDepositsCount,
+                tokenWithdrawAmount
+            );
```

Comment:
again we probably want to `setIndefiniteWhitelistStatus()` here as well.

Reply:
https://github.com/api3dao/airnode/pull/926/files/eee4256371ee411278e8f83c4bebbcdf7bd403cd#r824507853